codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ConstrainedShapeGeneratorCommon.kt

```diff
@@ -26,3 +28,5 @@ fun rustDocsTryFromMethod(
 fun rustDocsInnerMethod(inner: String) = "Returns an immutable reference to the underlying [`$inner`]."
 
 fun rustDocsIntoInnerMethod(inner: String) = "Consumes the value, returning the underlying [`$inner`]."
+
+fun Shape.idWithHashEscaped() = this.id.toString().replace("#", "##")
```

Comment:
Yes, I agree. This should be defined in a more central location. However, I am removing this for now. You make a good point that I should not be replacing '#' with '##' in a message since that is a requirement for `rustTemplate`. Therefore, the caller should do that; otherwise, the message would not be useful for any other purpose.